rundeckapp/grails-spa/packages/ui-trellis/src/library/components/storage/tests/KeyStoragePage.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { mount, VueWrapper } from "@vue/test-utils";
+import { ComponentPublicInstance } from "vue";
+import KeyStoragePage from "../KeyStoragePage.vue";
+import { Modal } from "uiv";
+jest.mock("@/library/rundeckService", () => ({
+  getRundeckContext: jest.fn().mockReturnValue({
+    rdBase: "mockRdBase",
+    projectName: "test-project",
+    rundeckClient: {
+      storageKeyGetMetadata: jest.fn().mockResolvedValue({}),
+      storageKeyDelete: jest.fn().mockResolvedValue({}),
+      projectList: jest.fn().mockResolvedValue([]),
+    },
+  }),
+  url: jest.fn().mockReturnValue({ href: "mockHref" }),
+}));
+interface KeyStoragePageData {
+  modalEdit: boolean;
+}
+type KeyStoragePageComponent = ComponentPublicInstance & KeyStoragePageData;
+const mountKeyStoragePage = async (
+  props = {},
+): Promise<VueWrapper<KeyStoragePageComponent>> => {
+  return mount(KeyStoragePage as unknown as KeyStoragePageComponent, {
+    props: {
+      project: "test-project",
+      readOnly: false,
+      allowUpload: true,
+      modelValue: "",
+      storageFilter: "",
+      ...props,
+    },
+    global: {
+      components: {
+        Modal,
+      },
+      mocks: {
+        $t: (msg) => msg,
+      },
+    },
+  }) as unknown as VueWrapper<KeyStoragePageComponent>;
+};
+describe("KeyStoragePage.vue", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("opens the editor when the open-editor event is emitted", async () => {
+    const wrapper = await mountKeyStoragePage();
+    const keyStorageView = wrapper.findComponent({ name: "KeyStorageView" });
+    expect(keyStorageView.exists()).toBe(true);
+    await keyStorageView.vm.$emit("update:modelValue", "openEditor");
```

Comment:
question: why `update:modelValue` is needed here?

Reply:
> question: why `update:modelValue` is needed here?

I was encountering a console warning when I didn't use it. 
```
 console.warn
    [Vue warn]: Invalid prop: type check failed for prop "uploadSetting". Expected Object, got Undefined  
      at <KeyStorageEdit project="test-project" root-path="keys/project/test-project" upload-setting=undefined  ... > 
      at <Modal id="storageuploadkey" ref="modalEdit" data-testid="modal-edit"  ... > 
      at <UiSocket section="edit-key-storage-providers" location="viewKeys" > 
      at <KeyStoragePage project="test-project" readOnly=false allowUpload=true  ... > 
```
Then I looked at the KeyStorageView component I found that it emits the event 
 emits: ["update:modelValue", "openEditor"],
 since this even is emitted in this component so by adding `update:modelValue` resolved the warning related to missing or incorrect prop.
 Same with the other test.
 Please let me know if there are other ways to fix this.

Reply:
```emits: ["update:modelValue", "openEditor"],```

that's the list of events that **can** be emitted by a component, but it's the parameters being passed to `this.$emit` calls that determine what is the event emitted and what is the payload.

There's no need to trigger `update:modelValue`. Please double check if any payload should be emitted alongside with the event `openEditor`